Automattic/jetpack
# Add Form block overlay

This PR improves the Form block selection.

Currently selecting the Form block or any of its inner blocks doesn't feel consistent. The inner blocks get selected upon any click, making it difficult to quickly select the Form block.

Before:
![before-form-select](https://user-images.githubusercontent.com/157240/195675348-1c66ffb7-4664-46f6-bdd8-34d73d7879ed.gif)


#### Changes proposed in this Pull Request:

This PR attempts to provide a better UX by adding a transparent overlay when neither the block or its inner blocks are selected.

When the overlay is present, it will capture the first click upon the Form block area, effectively selecting the  Form block.

NOTE: clicking on editables will override any attempt to select a particular block, the editable receives the focus and its block gets selected, this behavior is intended and this change can't/shouldn't override it.

After:
![after-form-select](https://user-images.githubusercontent.com/157240/195675406-a2115db1-9271-4bc9-b606-0e75b92d0d36.gif)

#### Other information:


#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

Checkout and insert a Form block on a post. Lose and regain focus on the Form block by clicking anywhere inside of it (with exception of editables, see NOTE above). The Form block should be easily selectable.
